scheduling-manager-workflows/src/main/java/com/amazonaws/blox/schedulingmanager/wrapper/ECSWrapperFactory.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may
+ * not use this file except in compliance with the License. A copy of the
+ * License is located at
+ *
+ *     http://aws.amazon.com/apache2.0/
+ *
+ * or in the "LICENSE" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+package com.amazonaws.blox.schedulingmanager.wrapper;
+
+import com.amazonaws.auth.AWSCredentialsProvider;
+import com.amazonaws.services.ecs.AmazonECSClient;
+import com.amazonaws.services.securitytoken.AWSSecurityTokenService;
+import lombok.AllArgsConstructor;
+import lombok.NonNull;
+
+@AllArgsConstructor
+public class ECSWrapperFactory implements WrapperFactory<ECSWrapper> {
```

Comment:
It might be more accurate to name this `CrossAccountAmazonECS` or similar, and to have it also implement the `AmazonECS` interface, perhaps using [Lombok's `@Delegate` annotation](https://projectlombok.org/features/experimental/Delegate).

Reply:
why do we need to reimplement the ecs interface?

Reply:
Single responsibility: If the `CrossAccountAmazonECS` class is only doing auth-decoration, that's a single responsibility. If it's also changing method signatures, it's also acting as an adapter, which is an orthogonal responsibility.

Considering we'd probably want an ECS adapter interface (something like  `ECSCluster(clusterID)` so that we don't have to pass cluster ID in for every call, for example), mixing in the auth concern with the adaptation would make this harder to test than if those were separate classes.

Reply:
We do have two separate classes don't we? The wrapper factory deals with auth/role, the wrapper with ECS calls.